salt/modules/heat.py

```diff
@@ -0,0 +1,870 @@
+
+    parameters
+        Parameter dict used to update the stack
+
+    poll
+        Poll and report events until stack complete
+
+    rollback
+        Enable rollback on update failure
+
+    timeout
+        Stack creation timeout in minutes
+
+    profile
+        Profile to build on
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' heat.update_stack name=mystack \\
+                 template_file=salt://template.yaml \\
+                 enviroment=salt://enviroment.yaml \\
+                 parameters="{"image": "Debian 8", "flavor": "m1.small"}" \\
+                 poll=5 rollback=False timeout=60 profile=openstack1
+
+    '''
+    h_client = _auth(profile)
+    ret = {
+        'result': True,
+        'comment': ''
+    }
+    if not name:
+        ret['result'] = False
+        ret['comment'] = 'Parameter name missing or None'
+        return ret
+    if not parameters:
+        parameters = {}
+    if template_file:
+        template_tmp_file = salt.utils.mkstemp()
+        tsfn, source_sum, comment_ = __salt__['file.get_managed'](
+            name=template_tmp_file,
+            template=None,
+            source=template_file,
+            source_hash=None,
+            user=None,
+            group=None,
+            mode=None,
+            saltenv='base',
+            context=None,
+            defaults=None,
+            skip_verify=False,
+            kwargs=None)
+
+        template_manage_result = __salt__['file.manage_file'](
+            name=template_tmp_file,
+            sfn=tsfn,
+            ret=None,
+            source=template_file,
+            source_sum=source_sum,
+            user=None,
+            group=None,
+            mode=None,
+            saltenv='base',
+            backup=None,
+            makedirs=True,
+            template=None,
+            show_changes=False,
+            contents=None,
+            dir_mode=None)
+        if template_manage_result['result']:
+            with salt.utils.fopen(template_tmp_file, 'r') as tfp_:
+                tpl = tfp_.read()
+                salt.utils.safe_rm(template_tmp_file)
+                try:
+                    if isinstance(tpl, six.binary_type):
+                        tpl = tpl.decode('utf-8')
+                    template = _parse_template(tpl)
+                except ValueError as ex:
+                    ret['result'] = False
+                    ret['comment'] = 'Error parsing template {0}'.format(ex)
+        else:
+            ret['result'] = False
+            ret['comment'] = "Can't open template: {0} {1}".format(template_file, comment_)
+    else:
+        ret['result'] = False
+        ret['comment'] = "Can't open template"
+    if ret['result'] is False:
+        return ret
+
+    kwargs = {}
+    kwargs['template'] = template
+    try:
+        h_client.stacks.validate(**kwargs)
+    except Exception as ex:  # pylint: disable=W0703
+        log.error("Template not valid {0}".format(ex))
```

Comment:
You can use `log.exception` here for better logging when displaying an exception. There are some examples sprinkled throughout the codebase.